xla/service/gpu/stream_attribute_annotator.cc

```diff
@@ -120,6 +120,13 @@ absl::StatusOr<bool> WrapIntoFusionAndAnnotateStreamAttributes(
       fusion_instruction->fused_instructions_computation(),
       absl::StrCat("wrapped_", wrapped_opcode, "_computation"));
   if (module->has_schedule()) {
+    // Update the scheduling names of the fusion and its root instruction
+    // to match their newly assigned instruction names during creation.
+    fusion_instruction->set_metadata_scheduling_name(
+        std::string(fusion_instruction->name()));
+    HloInstruction* root =
+        fusion_instruction->called_computations()[0]->root_instruction();
+    root->set_metadata_scheduling_name(std::string(root->name()));
```

Comment:
maybe set_metadata_scheduling_name method can be changed to take a absl::string_view as argument?

Reply:
set_metadata_scheduling_name() is declared and defined in hlo_instruction.h. If I change it, I'm afraid the change will be too big as at least another pass SchedulingInstructionAnnotator also uses it.

Reply:
I checked and it is only used by that other pass. Anyway, this is a pre-existing issue, so I will take care of it separately.

Reply:
Ah, I see you also did that change, thank you very much :)